metron-interface/metron-alerts/src/app/service/save-search.service.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Injectable, Inject, } from '@angular/core';
+import {Observable} from 'rxjs/Rx';
+import {Http} from '@angular/http';
+import {Subject} from 'rxjs/Subject';
+
+import {IAppConfig} from '../app.config.interface';
+import {APP_CONFIG} from '../app.config';
+import {ALERTS_SAVED_SEARCH, ALERTS_RECENT_SEARCH, NUM_SAVED_SEARCH} from '../utils/constants';
+import {QueryBuilder} from '../model/query-builder';
+import {SaveSearch} from '../model/save-search';
+import {ColumnMetadata} from '../model/column-metadata';
+
+@Injectable()
+export class SaveSearchService {
+
+  queryBuilder: QueryBuilder;
+  tableColumns: ColumnMetadata[];
+
+  private loadSavedSearch = new Subject<SaveSearch>();
+  loadSavedSearch$ = this.loadSavedSearch.asObservable();
+  defaultHeaders = {'Content-Type': 'application/json', 'X-Requested-With': 'XMLHttpRequest'};
+
+  constructor(private http: Http, @Inject(APP_CONFIG) private config: IAppConfig) {
+  }
+
+  deleteRecentSearch(saveSearch: SaveSearch): Observable<{}> {
+    return Observable.create(observer => {
+      let recentSearches: SaveSearch[] = [];
+      try {
+        recentSearches = JSON.parse(localStorage.getItem(ALERTS_RECENT_SEARCH));
+        recentSearches = recentSearches.filter(search => search.name !== saveSearch.name);
+      } catch (e) {}
+
+      localStorage.setItem(ALERTS_RECENT_SEARCH, JSON.stringify(recentSearches));
+
+      observer.next({});
+      observer.complete();
+
+    });
+  }
+
+  deleteSavedSearch(saveSearch: SaveSearch): Observable<{}> {
+    return Observable.create(observer => {
+      let savedSearches: SaveSearch[] = [];
+      try {
+        savedSearches = JSON.parse(localStorage.getItem(ALERTS_SAVED_SEARCH));
```

Comment:
@cestella yeah, I would like to move to REST based service ASAP. My thought was indirection would be good if we plan to have both localstorage and rest based storage as available options. The idea was to replace all calls to localstorage with rest calls in which case indirection would be redundant.
I am not heavily opinionated on this If you see some value in having indirection i will do that :)